shakenfist/external_api/app.py

```diff
@@ -608,6 +608,26 @@ def post(self, name=None, cpus=None, memory=None, network=None,
         # We need to sanitise the name so its safe for DNS
         name = re.sub(r'([^a-zA-Z0-9_\-])', '', name)
 
+        # Sanity check
+        if not disk:
+            return error(400, "Instance must specify at least one disk")
+        for d in disk:
+            if not isinstance(d, dict):
+                return error(400, "Disk specification should contain JSON objects")
+            if not all (k in d for k in ('size', 'base', 'bus', 'type')):
+                return error(400, "Disk specification is missing parameters: %s" % d)
+
+        if network:
+            for n in network:
+                if not isinstance(n, dict):
+                    return error(400,
+                        "Network specification should contain JSON objects")
+
+                net_params = ('network_uuid', 'address', 'macaddress', 'model')
+                if not all (k in n for k in net_params):
```

Comment:
Similarly here, some of these are optional...

* network_uuid: definitely required
* address: if not set we pick one for you
* macaddress: if not set we randomly generate one
* model: if not set you get the default from config